src/prompts.ts

```diff
@@ -73,339 +78,378 @@ export function readProviderPromptMap(
-  {
-    role: 'system',
-    content: `You are comparing multiple pieces of text to see which best fits the following criteria: {{criteria}}
+/**
+ * Processes a string as a literal prompt.
+ * @param {RawPrompt} prompt - The raw prompt data.
+ * @returns {Prompt[]} Array of prompts created from the string.
+ */
+function processString(prompt: RawPrompt): Prompt[] {
+  return [
+    {
+      raw: prompt!.raw as string,
+      label: `${prompt.source ? `${prompt.source}: ` : ''}${prompt.raw}`,
+    },
+  ];
+}
 
-Here are the pieces of text:
+/**
+ * Processes a raw prompt based on its content type and path.
+ * @param {RawPrompt} prompt - The raw prompt data.
+ * @param {string} basePath - Base path for file resolution.
+ * @returns {Promise<Prompt[]>} Promise resolving to an array of processed prompts.
+ */
+export async function processPrompt(
+  prompt: RawPrompt,
+  basePath: string = '',
+  maxRecursionDepth: number = 1,
+): Promise<Prompt[]> {
+  invariant(
+    typeof prompt.raw === 'string',
+    `prompt.raw must be a string, but got ${JSON.stringify(prompt.raw)}`,
+  );
+  if (!maybeFilePath(prompt.raw)) {
+    // literal prompt
+    return processString(prompt);
+  }
 
-{% for output in outputs %}
-<Text index="{{ loop.index0 }}">
-{{ output }}
-</Text>
-{% endfor %}
+  let promptPath = path.join(basePath, prompt.raw);
+  if (promptPath.includes('file:')) {
+    promptPath = promptPath.split('file:')[1];
+  }
 
-Output the index of the text that best fits the criteria. You must output a single integer.`,
-  },
-]);
+  const {
+    extension,
+    functionName,
+    isDirectory,
+  }: {
+    extension: string;
+    functionName?: string;
+    isDirectory: boolean;
+  } = parsePathOrGlob(promptPath);
+
+  if (isDirectory && maxRecursionDepth > 0) {
+    const globbedPath = globSync(promptPath.replace(/\\/g, '/'), {
+      windowsPathsNoEscape: true,
+    });
+    logger.debug(
+      `Expanded prompt ${prompt.raw} to ${promptPath} and then to ${JSON.stringify(globbedPath)}`,
+    );
+    const globbedFilePaths = globbedPath.flatMap((globbedPath) => {
+      const filenames = fs.readdirSync(globbedPath);
+      return filenames.map((filename) => path.join(globbedPath, filename));
+    });
+    const prompts: Prompt[] = [];
+    for (const globbedFilePath of globbedFilePaths) {
+      const processedPrompts = await processPrompt(
+        { raw: globbedFilePath },
+        basePath,
+        maxRecursionDepth - 1,
+      );
+      prompts.push(...processedPrompts);
+    }
+    return prompts;
+  }
+  if (extension === '.jsonl') {
+    return processJsonlFile(promptPath, prompt.label, prompt.raw);
+  }
+  if (extension === '.txt') {
+    return processTxtFile(promptPath, prompt);
+  }
+  if (extension === '.py') {
+    return processPythonFile(promptPath, prompt, functionName, prompt.raw);
+  }
+  if (['.js', '.cjs', '.mjs'].includes(extension)) {
+    return processJsFile(promptPath, functionName);
+  }
+  return [];
```

Comment:
if we make it this far, does this mean the glob matches nothing OR it's an unsupported extension?  In that case should we throw?

also, let's keep this outside this PR but seems like we should support yaml given support for yaml imports elsewhere.